reverse-proxy.md

```diff
@@ -44,9 +44,14 @@ All examples below will use port `11000` as `APACHE_PORT`. This port will be exp
 
     <summary>On the same server in a Docker container</summary>
 
-    For this setup, you can use as target `host.docker.internal:$APACHE_PORT` instead of `localhost:$APACHE_PORT`. **⚠️ Important:** In order to make this work on Docker for Linux, you need to add `--add-host=host.docker.internal:host-gateway` to the docker run command of your reverse proxy container or `extra_hosts: ["host.docker.internal:host-gateway"]` in docker compose (it works on Docker Desktop by default).
+    For this setup, the reverse-proxy container and the nextcloud containers (specifically the `nextcloud-aio-mastercontainer` and `nextcloud-aio-apache`) need to be connected to the same bridge network. This can be achieved in one of these two ways:
 
-    Another option (actually the recommended way) in this case is to use `--network host` option (or `network_mode: host` for docker-compose) as setting for the reverse proxy container to connect it to the host network. If you are using a firewall on the server, you need to open ports 80 and 443 for the reverse proxy manually. By doing so, the default sample configurations that point at `localhost:$APACHE_PORT` should work without having to modify them.
+    * Reverse Proxy container is connected to `nextcloud-aio` network, either by setting it up under the same service (same docker compose file) or as a secondary external network in the reverse proxy container service.
+    * The front-facing nextcloud containers (`nextcloud-aio-mastercontainer` and `nextcloud-aio-apache`) are connected to the docker network of the reverse proxy container by specifying env variable `APACHE_CONTAINER_ADDITIONAL_NETWORK`.
+
+    After this the reverse-proxy can utilize Docker DNS networking to access nextcloud by container names at `http://nextcloud-aio-apache:$APACHE_PORT` and `https://nextcloud-aio-mastercontainer:8080`.
+
+    A third alternative is to utilize host networking instead of docker bridge networking: Specify `--network host` option (or `network_mode: host` for docker-compose) as setting for the reverse proxy container to connect it to the host network. If you are using a firewall on the server, you need to open ports 80 and 443 for the reverse proxy manually.  By doing so, the default sample configurations that point at `localhost:$APACHE_PORT` should work without having to modify them.
```

Comment:
I am not satisfied by the formatting here.

you start an enumeration but are not using that for all the options.

Also, it is not the right place to talk about the `https://nextcloud-aio-mastercontainer:8080` here. imho.

Addiitonally, this setting is only going to work if one uses `SKIP_DOMAIN_VALIDATION: true`. So this needs to be accounted for.

Reply:
I am mentioning skip-domain-validation because people would first point at the domaincheck container and after the validation point at the apache container to make the validation work...